cloud/data_source_apikey.go

```diff
@@ -110,6 +113,11 @@ func dataSourceApiKey() *schema.Resource {
 				Computed:    true,
 				Description: descriptions["revoked_at"],
 			},
+			"principal_name": {
```

Comment:
Shouldn't the `streamnative_service_account` resource/data source need to support this as the principal name is the property of a service account, not an API key?

Reply:
On streamnative cloud, each apikey is bound to a service account, and multiple apikey objects can be generated under each service account
The apikey under a service account and the service account share the same `principal_name`, which is used as a role to grant permissions in the pulsar cluster.

The apikey is a token in jwt format.
In order to connect to the pulsar cluster, the service account will also need to exchange a jwt-formatted token on the client side in advance.
The only difference between apikey-based tokens and service account tokens is that apikey-generated tokens support revoke.

Reply:
I'll also add the `principal_name` to the service account in the next pr